lib/ddtrace/contrib/rack/patcher.rb

```diff
@@ -17,21 +17,25 @@ module Patcher
         module_function
 
         def patch
-          return true if patched?
+          unless patched?
+            require_relative 'middlewares'
+            @patched = true
+          end
 
-          require_relative 'middlewares'
-          @patched = true
+          if !@middleware_patched && get_option(:middleware_names) && get_option(:application)
```

Comment:
That sounds great, maybe we should add a deprecation warning with a condition such as:
```ruby
if get_option(:middleware_names) && !get_option(:application)
  log_deprecation_warning('you should pass the application or use Rails')  # change it with a proper message
end
```